tower-buffer/src/future.rs

```diff
@@ -35,36 +35,36 @@ where
     }
 }
 
-impl<T> Future for ResponseFuture<T>
+impl<T, A, B> Future for ResponseFuture<T>
 where
-    T: Future,
-    T::Error: Into<Error>,
+    T: Future<Output = Result<A, B>>,
+    B: Into<Error>,
 {
-    type Item = T::Item;
-    type Error = Error;
-
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        use self::ResponseState::*;
+    type Output = Result<A, Error>;
 
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
```

Comment:
I don't think you need the `<'_>` here

Reply:
Say more? `Context` always needs a life time right?

Reply:
You don't need to name the lifetime if you are just taking a reference to it. If you just remove it, it'll still compile just fine.